client/state/routing/middleware.js

```diff
@@ -3,50 +3,28 @@
  */
 import debugFactory from 'debug';
 import { isEmpty } from 'lodash';
-import page from 'page';
 
 /**
  * Internal dependencies
  */
-import {
-	getSavedPath,
-	savePath,
-} from 'lib/restore-last-path';
+import { savePath } from 'lib/restore-last-path';
 import { ROUTE_SET } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
-let hasInitialized = false;
 
 export const routingMiddleware = () => {
 	return ( next ) => ( action ) => {
-		if ( action.type !== ROUTE_SET || ! action.path || ! action.query ) {
+		if ( action.type !== ROUTE_SET ||
```

Comment:
completely ignorable, personal-opinion-based nit: 
What do you think to a format like so?:
```js
if (
    x === y,
    a !== b
) {
   return c;
}
```
My only rational for this is for slightly clearer indentation and boundaries between condition and logic - so FFTI.